### PR TITLE
add methods skeleton / create DTO

### PR DESCRIPTION
**Pull request by ifteakhor**

### CODE REVIEW

File: src/Controllers/IncomeController.cs [hunk 1]
```diff
@@ -1,13 +1,55 @@
...
+    [HttpPut]
+    public async Task<ActionResult<IncomeDto>> Update([FromBody] IncomeDto incomeDto)
+    {
+        // TODO: Update income
+        
+        throw new NotImplementedException();
+    }
+
+    [HttpDelete("{id}")]
+
```
Comment (ifteakhor) [inline src/Controllers/IncomeController.cs]:
```suggestion
```

File: src/Controllers/IncomeController.cs [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+    public async Task<ActionResult> Delete(int id)
     {
+        // TODO: Delete the income
+        
+        // maybe I will surround it by try catch
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs]: try catch - bad ❌
errors as values - good ❇️

File: src/Models/Income.cs [hunk 3]
```diff
@@ -2,11 +2,13 @@ namespace server.Models;
...
 {
-    public decimal Amount { get; set; }
+    public required decimal Amount { get; set; }
 
     public required DateTime Date { get; set; }
     
     public string? Note { get; set; }
 
-    public required string Method { get; set; }
+    public required string Category { get; set; }
```
Comment (veyxov) [inline src/Models/Income.cs]: Category should be a separate table

File: src/Context/MoniTrackContext.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+namespace server.Context;
+
+public class MoniTrackContext : DbContext
+{
+    public DbSet<Income> Incomes { get; set; } = null!;
+    public DbSet<Method> Method { get; set; } = null!;
+    
+    public string DbPath { get; }
+   
+    public MoniTrackContext()
```
Comment (veyxov) [inline src/Context/MoniTrackContext.cs:13]: There's no need for all of this.

File: src/Context/MoniTrackContext.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        var folder = Environment.SpecialFolder.LocalApplicationData;
+        var path = Environment.GetFolderPath(folder);
+        DbPath = System.IO.Path.Join(path, "blogging.db");
+    }
+    
+    protected override void OnConfiguring(DbContextOptionsBuilder options)
+        => options.UseSqlite($"Data Source={DbPath}");
```
Comment (veyxov) [inline src/Context/MoniTrackContext.cs:21]: `Data Source = app.db` will do the trick for now.

File: src/Controllers/IncomeController.cs [hunk 6]
```diff
@@ -1,11 +1,21 @@
...
 namespace Server.Controllers;
 
 [Route("api/[controller]")]
 public class IncomeController : ApiControllerBase
 {
+    private MoniTrackContext _context; 
+    
+    public IncomeController()
+    {
+        _context = new MoniTrackContext();
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs:16]: Please use [DI](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-7.0)

File: src/Controllers/IncomeController.cs [hunk 7]
```diff
@@ -15,19 +25,34 @@ public async Task<ActionResult<IEnumerable<IncomeDto>>> GetAll()
     }
     
     [HttpGet("{id}")]
-    public async Task<ActionResult<IncomeDto>> GetIncomeInfoById(int id)
+    public async Task<ActionResult<Income>> GetIncomeInfoById(int id)
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs:28]: Returning a database model, with many relations is problematic.
Circular dependency issues can occur.

File: src/Controllers/IncomeController.cs [hunk 8]
```diff
@@ -15,19 +25,34 @@ public async Task<ActionResult<IEnumerable<IncomeDto>>> GetAll()
...
+        return Ok(income);
     }
 
     [HttpPost]
     public async Task<ActionResult<IncomeDto>> Create([FromBody] IncomeDto incomeDto)
     {
-        // TODO: Create income
-        
-        throw new NotImplementedException();
+        Income income = new Income()
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs:43]: We can use something like [mapster](https://github.com/MapsterMapper/Mapster)

File: src/Dtos/IncomeDto.cs [hunk 9]
```diff
@@ -8,7 +8,7 @@ public class IncomeDto
     
     public string? Note { get; set; }
 
-    public required string Category { get; set; }
+    public string Category { get; set; } = null!;
```
Comment (veyxov) [inline src/Dtos/IncomeDto.cs:11]: This should be a table (not string, a concrete type)

**Review by veyxov — CHANGES_REQUESTED** [review]
Great job, take a look at the comments. Feel free to ask questions.

File: src/Controllers/IncomeController.cs
Comment (veyxov) [inline src/Controllers/IncomeController.cs:1]: Maybe rebase, in the future?

Comment (veyxov) [reply src/Controllers/IncomeController.cs:1]: merge commits clutter the history

File: src/Models/Income.cs (hunk 3 above)
Comment (ifteakhor) [reply src/Models/Income.cs]: done

File: src/Context/MoniTrackContext.cs (hunk 5 above)
Comment (ifteakhor) [reply src/Context/MoniTrackContext.cs:21]: done

File: src/Controllers/IncomeController.cs (hunk 6 above)
Comment (ifteakhor) [reply src/Controllers/IncomeController.cs:16]: done

File: src/Context/MoniTrackContext.cs (hunk 4 above)
Comment (ifteakhor) [reply src/Context/MoniTrackContext.cs:13]: removed

File: src/Dtos/IncomeDto.cs (hunk 9 above)
Comment (ifteakhor) [reply src/Dtos/IncomeDto.cs:11]: done

**Pull request merged**